### PR TITLE
Tweak to defer copy

### PR DESCRIPTION
**Pull request by coapacetic**
## What are you changing in this pull request and why?
I was working through an on demand video for learn.getdbt.com and was thinking about different ways to explain the defer command.  After reading through the docs and checking out how this works, I thought it might be helpful to contribute a rewording of the behavior here for people to follow the logic more closely.

Note: I incorrectly named the branch here 😬 

## Checklist
- [X] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [X] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [X] I have added checklist item(s) to this list for anything anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."
- [X] The content in this PR requires a dbt release note, so I added one to the [release notes page](https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes).
- [ ] Confirm technical accuracy of the proposed changes and that they are easy to follow.

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-tweak-clone-copy-dbt-labs.vercel.app/docs/cloud/about-cloud-develop-defer

### CODE REVIEW

File: website/docs/docs/cloud/about-cloud-develop-defer.md
Comment (mirnawong1) [inline website/docs/docs/cloud/about-cloud-develop-defer.md]:
```suggestion
2. If a development version doesn't exist, dbt uses the staging locations of parent relations based on metadata from the staging environment.
```

File: website/docs/docs/cloud/about-cloud-develop-defer.md
Comment (mirnawong1) [inline website/docs/docs/cloud/about-cloud-develop-defer.md]:
```suggestion
3. If both a development and staging version doesn't exist, dbt uses the production locations of parent relations based on metadata from the production environment.
```

File: website/docs/docs/cloud/about-cloud-develop-defer.md
Comment (mirnawong1) [inline website/docs/docs/cloud/about-cloud-develop-defer.md]:
```suggestion
**Note:** Passing the `--favor-state` flag will always resolve refs using production metadata, regardless of the presence of a development relation, skipping step #1.
```

**Review by mirnawong1 — APPROVED** [review]
great breakdown @coapacetic  and thanks for opening this pr to clarify it further! defer is tricky to understand and this certainly clarifies it! thank you!

**Comment by coapacetic** [comment]
Thanks @mirnawong1!!

@bethanyhipple-dbtlabs has some ideas on tweaking the *note:* that I'll let her jump in on before merge

**Pull request merged**
